### DISCUSSION TITLE
Retropie arcade machine joystick inverted

### DISCUSSION THREAD

**Discussion by maah001** [BlitterStudio/amiberry#1032] [Q&A] [answered]
Hi there. I've just finished building a Retropie arcade machine and all works fine apart from the fact that the joystick controls are reversed (i.e. left/right swapped as are up/down) when using Amiberry. The other emulators work fine using the emulation station controller configuration, just not the Amiga. Unfortunately, as the joysticks are built into the cabinet I can't just turn the joystick through 180° 🫤

Any guidance on how to fix this?
Many thanks.

**Comment by midwan** [comment, marked as answer]
There was a bug in Amiberry that got fixed recently, that is probably causing this. Some controllers report their axes inverted, and even though Amiberry detects and parses that correctly from the config file, it didn't respect that setting when reading the input.
This is already fixed in `master` and will be released as part of v5.5, coming soon.

**Comment by maah001** [comment]
Great. Thanks for the quick response.

Much appreciated.
